lib/index.js

```diff
@@ -93,29 +99,45 @@ class GraphQLComponent {
       return this._schema;
     }
 
-    const { importedTypes, importedResolvers, importedMocks, importedDirectives } = buildDependencyTree(this);
-
-    const typeDefs = mergeTypeDefs([...importedTypes]);
-    const resolvers = mergeResolvers([...importedResolvers]);
-    const schemaDirectives = importedDirectives;
-
-    const makeSchema = this._federation ? this.makeFederatedSchemaWithDirectives : makeExecutableSchema;
-
-    let schema = makeSchema({
-      typeDefs,
-      resolvers,
-      schemaDirectives
-    });
-
-    debug(`created ${this.name} schema`);
-
-    if (this._useMocks) {
-      debug(`adding mocks, preserveResolvers=${this._preserveResolvers}`);
-
-      schema = addMocksToSchema({schema, mocks: Object.assign({}, ...importedMocks), preserveResolvers: this._preserveResolvers});
+    if (this._imports.length > 0) {
+      // iterate through the imports and construct schema configuration objects
+      const subschemas = this._imports.map((imp) => {
+        const { component, exclude } = imp;
+        return {
+          schema: component.schema,
+          transforms: exclusions(exclude)
+        }
+      });
+
+      // construct a "gateway" schema from the schemas of imported
+      // components and this component's types/resolvers (if present)
+      this._schema = stitchSchemas({
+        subschemas,
+        typeDefs: this._types,
+        resolvers: this._resolvers,
+        schemaDirectives: this._directives
+      });
+    }
+    else {
```

Comment:
Does this mean if we have imports, we don't need to create a federated schema even though the federated option is set to true in the individual GraphQL Components?

Reply:
yep - agree with this

Reply:
implemented this in my latest push